s3direct/views.py

```diff
@@ -125,13 +131,19 @@ def generate_aws_v4_signature(request):
         resp = json.dumps({'error': 'S3 region config missing.'})
         return HttpResponseServerError(resp, content_type='application/json')
 
-    aws_credentials = get_aws_credentials()
-    if not aws_credentials.secret_key or not aws_credentials.access_key:
-        resp = json.dumps({'error': 'AWS credentials config missing.'})
-        return HttpResponseServerError(resp, content_type='application/json')
+    try:
+        access_key = request.session["aws_access_key_id"]
+        secret_key = request.session["aws_secret_access_key"]
+    except KeyError:
+        aws_credentials = get_aws_credentials()
+        access_key = aws_credentials.access_key
+        secret_key = aws_credentials.secret_key
```

Comment:
> should we store these in `request.session` at this point, if subsequent requests come in to this endpoint?

We can but it will not be executed in our use-case.

> but if for some reason it does, how does it play out?

If for some reason it does, then it will break in our case as it will use a different AWS credentials at that point given that we have multiple instances running. If the system that's using this is not in a load-balanced environment, then it can benefit from adding these values to the `session`. Adding that code should be pretty minor but I'm wondering if it's worth fixing given that there's a high chance executing this code in a load-balanced environment means getting `403s`?

I guess it will depend on what we're trying to achieve with this PR. Do we just want to solve "our" problem? Or do we want to do that _plus_ ensure that this won't cause issues when we create a PR for this into the original project? If we want to think of the latter, I feel like there are more things that we need to take care of. Things like tests, scenarios wherein sessions are disabled, systems that are not in load-balanced environment, etc.

Reply:
ok. so i think this essentially means that, if it hits this point and gets new creds, it wont matter anyway because the initial call to `get_upload_params` will have diff creds (or none) and the user will need to restart the upload from scratch. 

i'm ok with leaving it as it is.